python/lsst/jointcal/jointcal.py

```diff
@@ -498,7 +498,7 @@ def _build_ccdImage(self, data, associations, jointcalControl):
                                     data.wcs,
                                     data.visitInfo,
                                     data.bbox,
-                                    data.filter.getName(),
+                                    data.filterLabel.bandLabel,
```

Comment:
You say you need the filter to do filter-by-filter processing. Could you ever end up in a situation where you have to jointcal two different r-band filters?

Reply:
Good question. I think that's fine (especially since we're replacing jointcal's photometry mode with fgcmcal), but now I wonder.

Partly I did it this way because some of the existing testdata only knows `band`. Which is not the best reason, but I don't know a way to fix that without reprocessing testdata_jointcal.

Reply:
Noting for the record: the root problem (pre-FilterLabel calexps not having fully populated FilterLabels under `butler.get('calexp_filterLabel')` is being investigated in DM-28236.

Reply:
Thanks to your fix on DM-28236, I was able to switch this to use both band and physical here, and I cleaned up some variable names to make the uses much more obvious. Please have a look.